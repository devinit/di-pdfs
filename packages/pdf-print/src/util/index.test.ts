import 'jest';
import * as prettyFormat from 'pretty-format';
import {splitPdf} from '.';

describe('pdf print tests', () => {
    it('should split pdf', async () => {
        const pages = await splitPdf('./testData/brazil-old.pdf');
        expect(prettyFormat(pages)).toMatchSnapshot();
    }, 50000);
});
