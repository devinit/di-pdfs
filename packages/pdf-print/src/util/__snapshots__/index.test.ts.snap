// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pdf print tests should split pdf 1`] = `
"Socket {
  \\"_bytesDispatched\\": 0,
  \\"_events\\": Object {
    \\"_socketEnd\\": [Function onSocketEnd],
    \\"close\\": [Function anonymous],
    \\"end\\": [Function bound onceWrapper],
    \\"finish\\": [Function onSocketFinish],
  },
  \\"_eventsCount\\": 4,
  \\"_hadError\\": false,
  \\"_handle\\": Pipe {
    \\"onread\\": [Function onread],
    \\"owner\\": [Circular],
    \\"reading\\": true,
    \\"writeQueueSize\\": 0,
  },
  \\"_host\\": null,
  \\"_maxListeners\\": undefined,
  \\"_parent\\": null,
  \\"_pendingData\\": null,
  \\"_pendingEncoding\\": \\"\\",
  \\"_readableState\\": ReadableState {
    \\"awaitDrain\\": 0,
    \\"buffer\\": BufferList {
      \\"head\\": null,
      \\"length\\": 0,
      \\"tail\\": null,
    },
    \\"decoder\\": null,
    \\"defaultEncoding\\": \\"utf8\\",
    \\"destroyed\\": false,
    \\"emittedReadable\\": false,
    \\"encoding\\": null,
    \\"endEmitted\\": false,
    \\"ended\\": false,
    \\"flowing\\": null,
    \\"highWaterMark\\": 16384,
    \\"length\\": 0,
    \\"needReadable\\": true,
    \\"objectMode\\": false,
    \\"pipes\\": null,
    \\"pipesCount\\": 0,
    \\"readableListening\\": false,
    \\"reading\\": true,
    \\"readingMore\\": false,
    \\"resumeScheduled\\": false,
    \\"sync\\": false,
  },
  \\"_server\\": null,
  \\"_sockname\\": null,
  \\"_writableState\\": WritableState {
    \\"bufferProcessing\\": false,
    \\"bufferedRequest\\": null,
    \\"bufferedRequestCount\\": 0,
    \\"corked\\": 0,
    \\"corkedRequestsFree\\": Object {
      \\"entry\\": null,
      \\"finish\\": [Function bound onCorkedFinish],
      \\"next\\": null,
    },
    \\"decodeStrings\\": false,
    \\"defaultEncoding\\": \\"utf8\\",
    \\"destroyed\\": false,
    \\"ended\\": false,
    \\"ending\\": false,
    \\"errorEmitted\\": false,
    \\"finalCalled\\": false,
    \\"finished\\": false,
    \\"highWaterMark\\": 16384,
    \\"lastBufferedRequest\\": null,
    \\"length\\": 0,
    \\"needDrain\\": false,
    \\"objectMode\\": false,
    \\"onwrite\\": [Function bound onwrite],
    \\"pendingcb\\": 0,
    \\"prefinished\\": false,
    \\"sync\\": true,
    \\"writecb\\": null,
    \\"writelen\\": 0,
    \\"writing\\": false,
  },
  \\"allowHalfOpen\\": false,
  \\"connecting\\": false,
  \\"domain\\": null,
  \\"readable\\": true,
  \\"server\\": null,
  \\"writable\\": false,
  Symbol(asyncId): 157,
  Symbol(bytesRead): 0,
}"
`;
